odata-csdl/14 Vocabulary and Annotation.md

```diff
@@ -674,6 +674,44 @@ properties of singletons or entities in a particular entity set. These
 annotations override annotations on the properties or navigation
 properties targeted via the declaring structured type.
 
+External targeting is also possible for bound action and function overloads
+whose binding parameter is addressed via a certain path or has a certain type,
+as well as for parameters and return types thereof. The targeting expression then
+consists of two path expressions from the table above separated by a forward
+slash. The first path expression references one of the following model
+elements:
+
+Model element|means the binding parameter must be
+-------------|-----------------------------------
+Entity Set|an instance or collection of instances from the entity set
+Singleton|the singleton
+Navigation Property via container|addressed via the resource path to the navigation property
+Property via container|addressed via the resource path to the property
+Navigation Property via structured type|addressed by a path that traverses the structured type
+Property via structured type|addressed by a path that traverses the structured type
+Complex Type|an instance or collection of the type or a subtype thereof
+Entity Type|an instance or collection of the type or a subtype thereof
+
+A binding parameter is "addressed via a resource path" if the path expression addressing the
+binding parameter starts with the resource path, it may then continue with segments
+described in [OData-URL](#ODataURL), sections 4.6 through 4.12. If the resource path
+addresses a collection, the binding parameter can also be a single instance of that
+collection if a segment described in [OData-URL](#ODataURL), sections 4.9 and 4.10, follows.
```

Comment:
Does the annotation target
```
MySchema.MyContainer/MyEntitySet/MyAction
```
address the bound action
```
POST MyEntitySet/MyComposableFunction()/MyAction
```
even if the composable function returns a _different_ entity set?

Reply:
I would expect that it does _not_, instead I'd expect that annotations for target
`MySchema.MyContainer/MyOtherEntitySet/MySchema.MyAction` apply if function `MySchema.MyComposableFunction` returns entities from `MyOtherEntitySet`.

Reply:
> instead I'd expect that annotations for target
> `MySchema.MyContainer/MyOtherEntitySet/MySchema.MyAction` apply if function `MySchema.MyComposableFunction` returns entities from `MyOtherEntitySet`.

Which is the case according to the first table row:
Model element|means the binding parameter must be
-------------|-----------------------------------
Entity Set|an instance or collection of instances from the entity set

Reply:
But the target
```
MySchema.MyContainer/MyOtherEntitySet/MyContainmentNavigationProperty/MySchema.MyAction
```
would not apply if the composable function returned entities from the contained entity set.

Must we distinguish between containment and non-containment navigation properties? Currently we have only

Model element|means the binding parameter must be
-------------|-----------------------------------
Navigation Property via container|addressed via a resource path to the navigation property
Navigation Property via structured type|addressed via a resource path to the navigation property that traverses an instance or collection of the structured type

Reply:
If the function returns totally unrelated entities, we can't express that (yet) in $metadata.

If the function returns related entities and is defined with
~~~xml
EntitySetPath="__it/_otherEntity/MyContainmentNavigationProperty"
~~~
then the client could figure out look for annotations with target
~~~
MySchema.MyContainer/MyOtherEntitySet/MyContainmentNavigationProperty/MySchema.MyAction
~~~

That is asking a lot, and I feel we are down a very exotic rabbit hole.